src/pages/Storefront/Cart.js

```diff
@@ -16,17 +19,87 @@ import {
   Link,
   chakra,
 } from '@chakra-ui/react';
-import PreserveQueryParamsLink from 'components/storefront/PreserveQueryParamsLink/PreserveQueryParamsLink';
-import { PageContainer } from 'components/storefront/PageContainer/PageContainer';
-import CartSkeleton from 'components/storefront/Cart/Layout/CartSkeleton';
+
+const calculateDiscount = (discountApplications, lineItems) => {
+  let discountVal = 0;
+  let shippingCost = 'TBD';
+  let discountType = '';
+  if (discountApplications.length > 0) {
+    const appliedDiscount = discountApplications[0];
+    discountType = appliedDiscount.targetType;
+    if (discountType === 'SHIPPING_LINE') {
+      shippingCost = 'FREE';
+    } else if (discountType === 'LINE_ITEM') {
+      discountVal = lineItems
+        .reduce((acc, lineItem) => {
+          return lineItem.discountAllocations
+            ? acc +
+                lineItem.discountAllocations.reduce((acc2, discAlloc) => {
+                  return acc2 + parseFloat(discAlloc.allocatedAmount.amount);
+                }, 0)
+            : acc;
+        }, 0)
+        .toFixed(2);
+    }
+  }
+  return { discountVal, shippingCost, discountType };
+};
+
+const DiscountMsg = ({ discountApplications, discountSuccess, removeCoupon }) => {
```

Comment:
There probably is a better way of determining the success/error message based on the coupon code

Reply:
agree, idk how this works — but it works :D